avm2: Implement Serialization for ByteArrayObject

nivkner opened the pull request:
encountered the warning `Serialization is not implemented for class other than Object: ByteArrayObject` in "Super Crazy Guitar maniac Deluxe 4".

Lord-McSweeney commented:
Welcome, and thanks for your contribution! Could you add a test for the serialization? Information on making tests is at https://github.com/ruffle-rs/ruffle/blob/master/tests/README.md; you can make a new test called "`bytearray_serialization`".

nivkner commented:
added the tests, based on `tests/tests/swfs/avm2/amf_xml`.
added `extends Sprite` because flashplayer crashed with
```

(flashplayerdebugger:9507): GLib-GObject-CRITICAL **: 11:44:11.941: instance with invalid (NULL) class pointer

(flashplayerdebugger:9507): GLib-GObject-CRITICAL **: 11:44:11.941: g_signal_handlers_disconnect_matched: assertion 'G_TYPE_CHECK_INSTANCE (instance)' failed
```
when compiled with Apache Flex SDK using a container.
